MELA/fortran/mod_TopDecay.F90

```diff
@@ -100,6 +100,52 @@ SUBROUTINE TopDecay(Flavor,Mom,Spinor,TopHel)
 
 
 
+
+SUBROUTINE WDecay(Charge,Mom,WCurr)
```

Comment:
Do you think you could describe how this function is used?

Reply:
this function is for the decay of a W boson with polarization vector WCurr and momentum Mom.
It is needed for the spin-correlated decay of the W boson in tHW production (cf. TopDecay(Flavor,Mom,Spinor,TopHel))

Reply:
to clarify: this function is basically the W-decay part taken from the TopDecay subroutine, thus it appears in mod_TopDecay. For tHW both the top decay and the W decay are needed in terms of the spinors/polarization vectors used in the stable production.